framework/include/GameBoard.h

```diff
@@ -0,0 +1,49 @@
+#ifndef GAMEBOARD_H
+#define GAMEBOARD_H
+
+#include "game.h"
+#include <string>
+
+namespace Game {
+
+// GameBoard class
+// Used to store and interface with the board state
+// Implementation is defined by the user depending on the game
+class GameBoard {
+public:
+
+    // Note: The following functions are constant for each game, do not modify
+    GameBoard() = default;
+    ~GameBoard() = default;
+    
+    // Return the board state
+    Game::boardstate_t* getBoardState() { return &boardState; }
```

Comment:
Is the C-style pointer intentionally used here over std::shared_ptr?

Reply:
It should be a std::shared_ptr, yes. I just removed the function for now. We will have to determine whether we want to return a pointer or do a copy via pass by reference once we actually need said function. Currently it isn't used anywhere.